### ISSUE TITLE
Training data for Dice

### ISSUE THREAD

**Issue by Saladino93** [interpretml/DiCE#189]
Something is unclear to me is how data is handled with Dice.

So, ideally we should have some training data over which we train the explorer, then a separate data set over which to apply it, am I right?

So why in several examples I see the dice_ml.Data take all the available data, then apply to the same data the counterfactual engine?

**Comment by gaugup** [comment]
@Saladino93 yes dice-ml explainers can be trained with just training data and then counterfactuals can be generate on any test set. 

Which  notebooks are you refering to?

Regards,

**Comment by Saladino93** [comment]
Thanks @gaugup . This was a notebook for example where dice Data are all of the data https://github.com/interpretml/DiCE/blob/master/docs/source/notebooks/DiCE_getting_started.ipynb

**Comment by gaugup** [comment]
Thanks @Saladino93. We can update the notebook to use samples from test set to generate the counterfactuals. But the choice of train/test to generate the counterfactuals should not impact the counterfactual generation as such. Does that help with you original question?

Regards,

**Comment by Saladino93** [comment]
@gaugup makes sense, thanks!

**Comment by gaugup** [comment]
@Saladino93, the above PR is merged now. The example notebook should use test data for generation of CFs. Can we close this issue?

**Comment by Saladino93** [comment]
Yes, I can close it, thanks!

**Comment by Saladino93** [comment]
Just noted you write:

 d = dice_ml.Data(dataframe=dataset, continuous_features=['age', 'hours_per_week'], outcome_name='income')

but I think dataset = training data, not all of the data.

Thanks.

**Comment by gaugup** [comment]
Sorry about my understanding of the problem. I changed the dataset for dice-ml data interface to a train dataset. PR is here:- https://github.com/interpretml/DiCE/pull/198

**Comment by gaugup** [comment]
@Saladino93, the above PR is merged now. Can we close this issue?

Regards,

**Comment by gaugup** [comment]
closing this issue since the related PR is merged.
